src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -0,0 +1,310 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Types for holding quantities with units."""
+
+import math
+from typing import Self
+
+
+class Quantity:
+    """A quantity with a unit."""
+
+    _base_value: float
+    """The value of this quantity in the base unit."""
+
+    _exponent_unit_map: dict[int, str]
+    """A mapping from the exponent of the base unit to the unit symbol."""
+
+    def __init__(self, value: float, *, exponent: int = 0) -> None:
```

Comment:
Based on the words used in https://en.wikipedia.org/wiki/Metric_prefix, we could change
- `exponent` to `base10_multiple` or similar
- `_exponent_unit_map` to `metric_prefix_map`

Reply:
any opinions on this? @matthias-wende-frequenz @llucax

Reply:
I don't an opinion in particular, `exponent` sounds more natural and `base10_multiple` sounds more specific. I don't have a strong preference.

If we are going with the metric system explicitly, I would make `Quantity` provide the map instead of allowing the subclasses to use any arbitrary "exponent" (as I said in some other comment). It should be enough to specify the unit and then all the sub-units can be generated from that and the metric specification.